templates/r10k-code.cronjob.yaml

```diff
@@ -91,7 +91,10 @@ spec:
             {{ toYaml .Values.nodeSelector.commonStoragePods | nindent 14 }}
           {{- end }}
         {{- end }}
-        {{- if .Values.affinity }}
+        {{- if .Values.r10k.affinity }}
```

Comment:
Why do we need a separate affinity just for the `r10k` job when there's an optional node selector exactly for that purpose?
--  https://github.com/Xtigyro/puppetserver-helm-chart/blob/master/values.yaml#L262

There's also optional affinity for all pods.
-- https://github.com/Xtigyro/puppetserver-helm-chart/blob/master/values.yaml#L268

Reply:
> Why do we need a separate affinity just for the `r10k` job when there's an optional node selector exactly for that purpose?

@Xtigyro that might be an option - will try as soon as I'm back to office on Jan 6.  Sorry in case of causing unnecessary trouble!

Reply:
Maybe I'm missing something put what we did was to specify that the r10k pod should always run on the same node as the puppetserver pod. As our nodes are changing from time to time we didn't want to specify a node/node label for the pods. So basically it doesn't matter on which node the pods run, it's just important that the r10k pod needs to run on the same as the puppetserver one.

Reply:
@tarioch @chwehrli 
Guys, thank you for your comments. I appreciate it! 

I would suggest to try the optional node selector here which has been mentioned in my first comment: https://github.com/Xtigyro/puppetserver-helm-chart/blob/master/values.yaml#L262

It goes with good explanation what it's aiming for. If that does not achieve what you are looking for - please comment here and we can discuss further.

```yaml
## Pod Configuration
##
## Node labels for pod assignment
nodeSelector:
  ## Common node labels for pod assignment
  allPods: {}
    # key: value
    #
  ## Node labels for pod assignment of "puppetserver" and "r10k" pods
  ## Note that the "puppetserver" and "r10k" components of the Puppet Server
  ## use common storage. That common storage requirement might impose
  ## certain pod assignment scenarios. It might be required the "puppetserver" and "r10k"
  ## components to sit in a common Cloud Provider's Availability Zone.
  ## For that matter separate node labels are made available for those if need be.
  commonStoragePods: {}
    # failure-domain.beta.kubernetes.io/zone: eu-central-1a
    # kubernetes.io/hostname: ip-10-0-6-168.eu-central-1.compute.internal
```

Reply:
At least in our setup, the default pvc can only be mounted by multiple pods of they are running on the same node. The easiest way to define this without having specific nodes or having to label nodes (as in our setup they are changing). Is to use a pod affinity for the r10k pod like

```
affinity:
         podAffinity:
           requiredDuringSchedulingIgnoredDuringExecution:
           - labelSelector:
               matchExpressions:
               - key: component
                 operator: In
                 values:
                 - puppetserver
             topologyKey: "kubernetes.io/hostname"
```

This ensures that the r10k pod always runs on the same node as the puppetserver pod and therefore can use the same pvc for the code sharing.
This can not be a global affinity as this should only be true for the r10k pod, everything else can run on any node. 
This does not require any special node labeling.

Reply:
@tarioch @chwehrli @slconley 
Thank you for the exhaustive reply - appreciated!
Sounds fair. Will think about it a bit more. 

Some code changes will be required but we might merge the PR and modify/tweak further, afterwards.